ironic/drivers/modules/drac/inspect.py

```diff
@@ -44,7 +93,261 @@ class DracRedfishInspect(redfish_inspect.RedfishInspect):
+                            {'mac_data': macs, 'node': node.uuid})
+        else:
+            LOG.warning("No NIC information discovered "
+                        "for node %(node)s", {'node': node.uuid})
+
+        return states.MANAGEABLE
+
+    def _get_pxe_dev_nics(self, task, system):
+        """Get a list of pxe device interfaces.
+
+        :param task: a TaskManager instance.
+        :param system: a Redfish System object that represents a node.
+        :returns: Returns list of pxe device interfaces.
+        """
+        pxe_dev_nics = []
+        pxe_params = ["PxeDev1EnDis", "PxeDev2EnDis",
+                      "PxeDev3EnDis", "PxeDev4EnDis"]
+        pxe_nics = ["PxeDev1Interface", "PxeDev2Interface",
+                    "PxeDev3Interface", "PxeDev4Interface"]
+
+        redfish_ip = task.node.driver_info.get('redfish_address')
+        redfish_username = task.node.driver_info.get('redfish_username')
+        redfish_password = task.node.driver_info.get('redfish_password')
+
+        if system.boot.mode == 'uefi':
+            for param, nic in zip(pxe_params, pxe_nics):
+                if system.bios.attributes[param] == 'Enabled':
+                    nic_id = system.bios.attributes[nic]
+                    mac_response = requests.get('%s%s%s' % (
+                        redfish_ip, _MAC_URL, nic_id), auth=(
+                        redfish_username, redfish_password), verify=False)
+                    mac_address = json.loads(mac_response.__dict__.get(
+                        '_content')).get('MACAddress')
+                    pxe_dev_nics.append(mac_address)
+
+        elif system.boot.mode == 'bios':
+            redfish_username = task.node.driver_info.get('redfish_username')
+            redfish_password = task.node.driver_info.get('redfish_password')
+
+            url = "%s%s" % (redfish_ip, _JOB_ID_URL)
+            job_response = requests.post(url, data=json.dumps(
+                _PAYLOAD), headers=_HEADERS, verify=False, auth=(
+                redfish_username, redfish_password))
+
+            if job_response.status_code != _JOB_RESPONSE_CODE:
+                LOG.debug("FAIL, status code not 202, code is: %(code), "
+                          "Error details: %(err)",
+                          {'code': job_response.status_code,
+                           'err': job_response.__dict__})
+                return pxe_dev_nics
+
+            job_id = job_response.__dict__['headers']['Location']
+            try:
+                job_id = re.search("JID_.+", job_id).group()
+            except AttributeError:
+                LOG.debug("FAIL: detailed error message: %(err)",
+                          {'err': job_response.__dict__['_content']})
+                return pxe_dev_nics
+            start_time = datetime.now()
+            while True:
+                current_time = (datetime.now() - start_time)
+                xml_data = requests.get('%s%s%s' % (
```

Comment:
Would you mind emailing me what you get back for xml_data?  I'm very curious to see what it is.  Thanks!

Reply:
I sent you an email, Done.